torch/_tensor_str.py

```diff
@@ -145,7 +145,13 @@ def format(self, value):
                 ret = ('{{:.{}f}}').format(PRINT_OPTS.precision).format(value)
         elif self.complex_dtype:
             p = PRINT_OPTS.precision
-            ret = '({{:.{}f}} {{}} {{:.{}f}}j)'.format(p, p).format(value.real, '+-'[value.imag < 0], abs(value.imag))
+            # format real and imaginary values according to type
+            real_val = '({{:.0f}}.    '.format(p).format(value.real) if value.real.is_integer() \
```

Comment:
```real_val = '({{:.0f}}.'.format(p).format(value.real) if value.real.is_integer() \```

Reply:
I think we should remove the extra white space

Reply:
Addressed in PR update.